cmake of tensorflowlite for aarch64 failed with EIGEN_STATIC_ASSERT

<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Source

source

### Tensorflow Version

master

### Custom Code

No

### OS Platform and Distribution

_No response_

### Mobile device

_No response_

### Python version

_No response_

### Bazel version

_No response_

### GCC/Compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current Behaviour?

```shell
My workspace has tensorflow master, last commit:
commit 21b567d26ca4cfa5c02260acd77b8b5f2fd57d84 (HEAD -> master, origin/master, origin/HEAD)
Author: Emilio Cota <ecg@google.com>
Date:   Thu Sep 29 15:14:11 2022 -0700

    Update CMakeLists to reflect the Arithmetic->Arith dialect rename

After fixing https://github.com/tensorflow/tensorflow/issues/58476 I installed flatbuff to a directory and set PATH to include that. Then I can continue to build for the shared tensorflowlite lib, according to https://www.tensorflow.org/lite/guide/build_cmake_arm
Pls refer to the log output for the error logs. I think this commit caused problem but I don't know how to fix: https://github.com/tensorflow/tensorflow/commit/09d49d9f6c9c13679ab8181d1b15480ea8a88da1
```


### Standalone code to reproduce the issue

```shell
ARMCC_PREFIX=... (set to our own toolchain)
ARMCC_FLAGS="-funsafe-math-optimizations"

cmake -DCMAKE_C_COMPILER=${ARMCC_PREFIX}gcc \
  -DCMAKE_CXX_COMPILER=${ARMCC_PREFIX}g++ \
  -DCMAKE_C_FLAGS="${ARMCC_FLAGS}" \
  -DCMAKE_CXX_FLAGS="${ARMCC_FLAGS}" \
  -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
  -DCMAKE_SYSTEM_NAME=Linux \
  -DCMAKE_SYSTEM_PROCESSOR=aarch64 \
  ../tensorflow/lite/  -DTFLITE_ENABLE_GPU=ON
cmake --build . -j
```


### Relevant log output

```shell
/opt/workspace/tensorflow/tensorflow/lite/kernels/internal/optimized/multithreaded_conv.h:173:65:   required from here
/opt/workspace/tensorflow/tensorflow/core/kernels/eigen_spatial_convolutions-inl.h:1068:27: error: static assertion failed: YOU_MADE_A_PROGRAMMING_MISTAKE
 1068 |   EIGEN_STATIC_ASSERT((nr == 4), YOU_MADE_A_PROGRAMMING_MISTAKE)
      |                       ~~~~^~~~~
/opt/workspace/tensorflow/tflite_build/eigen/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/StaticAssert.h:26:50: note: in definition of macro 'EIGEN_STATIC_ASSERT'
   26 | #define EIGEN_STATIC_ASSERT(X,MSG) static_assert(X,#MSG);
      |                                                  ^
```
</details>

--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -156,7 +156,7 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
 endif()
 # Additional compiler flags used when compiling TF Lite.
 set(TFLITE_TARGET_PUBLIC_OPTIONS "")
-set(TFLITE_TARGET_PRIVATE_OPTIONS "-DEIGEN_NEON_GEBP_NR=4")
+set(TFLITE_TARGET_PRIVATE_OPTIONS "")
 set(TFLITE_TARGET_PRIVATE_DEFINITIONS "")
 # Additional library dependencies based upon enabled features.
 set(TFLITE_TARGET_DEPENDENCIES "")
@@ -373,7 +373,7 @@ if(TFLITE_ENABLE_GPU)
      target_link_libraries(${lib_name})
    endforeach()
 endif()
-  list(APPEND TFLITE_TARGET_PUBLIC_OPTIONS "-DCL_DELEGATE_NO_GL" "-DEGL_NO_X11")
+  list(APPEND TFLITE_TARGET_PUBLIC_OPTIONS "-DCL_DELEGATE_NO_GL" "-DEGL_NO_X11" "-DEIGEN_NEON_GEBP_NR=4")
   list(APPEND TFLITE_TARGET_DEPENDENCIES
     absl::any
     absl::flat_hash_map

